catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/notes/NoteUtil.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.notes;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.boon.json.JsonFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.types.Core;
+
+public class NoteUtil {
+
+    private NoteUtil() { }
+
+    /**
+     * Extracts the data required by the user from the note {@link Metacard} object.
+     * If any required fields are null the return object will also be null.
+     *
+     * @param metacard - the metacard object to abstract response data from
+     * @return The JSON String
+     **/
+    public static String getResponseNoteJson(Metacard metacard) {
+        Map<String, String> requiredAttributes = new HashMap<>();
+        if (metacard.getId() != null) {
+            requiredAttributes.put("id", metacard.getId());
+        } else {
+            return null;
+        }
+        if (metacard.getAttribute(Note.PARENT_ID) != null) {
+            requiredAttributes.put("parent",
+                    (String) metacard.getAttribute(Note.PARENT_ID).getValue());
+        } else {
+            return null;
+        }
+        if (metacard.getCreatedDate() != null) {
+            requiredAttributes.put("created", metacard.getCreatedDate().toString());
+        }
+        if (metacard.getModifiedDate() != null) {
+            requiredAttributes.put("modified", metacard.getModifiedDate().toString());
+        }
+        if (metacard.getAttribute(Note.COMMENT) != null) {
+            requiredAttributes.put("note",
+                    (String) metacard.getAttribute(Note.COMMENT).getValue());
+        } else {
+            return null;
+        }
+        if (metacard.getAttribute(Core.METACARD_OWNER) != null) {
+            requiredAttributes.put("owner",
+                    (String) metacard.getAttribute(Core.METACARD_OWNER).getValue());
+        } else {
+            return null;
+        }
+        return JsonFactory.create().writeValueAsString(requiredAttributes);
```

Comment:
✏️  I would recommend doing the encoding to a specific format (in this case JSON) outside of this method so we do not couple the encoding format with the actual method logic. So instead just return the map and have it encoded higher up in the application logic

Reply:
done